sso/resources/org.jaggeryjs.modules.sso/src/main/java/org/jaggeryjs/modules/sso/common/util/Util.java

```diff
@@ -100,19 +102,13 @@ public static XMLObject buildXMLObject(QName objectQName)
      */
     public static String createID() {
 
-        byte[] bytes = new byte[20]; // 160 bits
-        random.nextBytes(bytes);
-
-        char[] chars = new char[40];
-
-        for (int i = 0; i < bytes.length; i++) {
-            int left = (bytes[i] >> 4) & 0x0f;
-            int right = bytes[i] & 0x0f;
-            chars[i * 2] = charMapping[left];
-            chars[i * 2 + 1] = charMapping[right];
+        try {
+            SecureRandomIdentifierGenerator generator = new SecureRandomIdentifierGenerator();
+            return generator.generateIdentifier();
+        } catch (NoSuchAlgorithmException e) {
+            log.warn("Error while building Secure Random ID");
         }
-
-        return String.valueOf(chars);
+        return null;
```

Comment:
It will only be executed if the try block fails to create the ID. 
Explained in https://github.com/wso2/jaggery-extensions/pull/84